library/nsxt_upgrade_postchecks.py

```diff
@@ -0,0 +1,169 @@
+#
+# Copyright 2019 VMware, Inc.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,
+# BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nsxt_upgrade_prechecks
+short_description: 'Execute pre-upgrade checks'
+description: "Run pre-defined checks to identify potential issues which can be 
+              encountered during an upgrade or can cause an upgrade to fail. The results 
+              of the checks are added to the respective upgrade units aggregate-info. The 
+              progress and status of operation is part of upgrade status summary of 
+              individual components."
+version_added: '2.7'
+author: 'Kommireddy Akhilesh'
+options:
+    hostname:
+        description: 'Deployed NSX manager hostname.'
+        required: true
+        type: str
+    username:
+        description: 'The username to authenticate with the NSX manager.'
+        required: true
+        type: str
+    password:
+        description: 'The password to authenticate with the NSX manager.'
+        required: true
+        type: str
+    state:
+        choices:
+            - present
+            - absent
+        description: "State can be either 'present' or 'absent'.
+                      'present' is used to run pre upgrade checks.
+                      'absent' is used to abort preupgrade checks."
+        required: true   
+'''
+
+EXAMPLES = '''
+- name: Runs and aborts pre-upgrade checks
+  nsxt_upgrade_prechecks:
+      hostname: "10.192.167.137"
+      username: "admin"
+      password: "Admin!23Admin"
+      validate_certs: False
+      state: 'present'
+'''
+
+RETURN = '''# '''
+
+import json, time
+from csv import reader
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware_nsxt import vmware_argument_spec, request
+from ansible.module_utils.common_utils import get_attribute_from_endpoint, clean_and_get_params, get_upgrade_orchestrator_node
+from ansible.module_utils._text import to_native
+
+def wait_for_pre_upgrade_checks_to_execute(manager_url, endpoint, mgr_username,
```

Comment:
I think, this would be post_upgrade, this module will fail at compile time itself. Don't wait for QE to raise bug, better test all module yourself.

Reply:
Done.